### PR TITLE
Tow 175 implement task

### PR DESCRIPTION
**Pull request by JordanBandur**
## Migrations Required?
YES :white_check_mark:

## Summary (a few sentences describing what this PR is doing) 
Implement the Task entity, including model, dao, serializers, views, URLs, and admin configuration. Ensure the tasks can be created, updated, retrieved, and deleted via the API and are visible in the Django admin interface.

## Checks
-  [x] I have ran all the tests locally, and they all pass

## Related PRs
N/A

## Impacted Areas in Application
List the areas in the app this PR is going to affect 
- services.py
- types.py
- models.py
- serializers.py
- dao.py
- views.py
- urls.py
- admin.py

## Learnings
Practice adding a new entity to an existing database from start to finish (integrating the entity with it's relationships)

### CODE REVIEW

**Review by GagenvirGill — APPROVED** [review]
The implementation looks great! However, I'm not too sure what the task entity is and its relation to the other models, its not on the ERD on notion.

**Comment by JordanBandur** [comment]
> The implementation looks great! However, I'm not too sure what the task entity is and its relation to the other models, its not on the ERD on notion.

I believe Ryan is in the process of updating the ERD. The Task Entity is part of Evan's updated mock-ups.

File: townhall/myapi/dao.py [hunk 1]
```diff
@@ -251,3 +253,44 @@ def filtered_organization(
             filters["location__icontains"] = filtered_organization_data.location
 
         return Organization.objects.filter(**filters)
+
+class TaskDao:
+
+    @staticmethod
+    def get_all_tasks():
```
Comment (ansel-rangers11) [inline townhall/myapi/dao.py:260]: please add type hinting to all these methods

File: townhall/myapi/dao.py [hunk 2]
```diff
@@ -251,3 +253,44 @@ def filtered_organization(
...
+
+        return Task.objects.all()
+
+    @staticmethod
+    def get_task_by_id(task_id: int):
+
```
Comment (ansel-rangers11) [inline townhall/myapi/dao.py:266]: not sure why theres whitespace after every method definition, it shouldnt be there generally

File: townhall/myapi/dao.py [hunk 3]
```diff
@@ -251,3 +253,44 @@ def filtered_organization(
...
+        except ObjectDoesNotExist:
+            return None
+
+    @staticmethod
+    def create_task(task_data: dict):
+
+        return Task.objects.create(**task_data)
+
+    @staticmethod
+    def update_task(task_id: int, task_data: dict):
```
Comment (ansel-rangers11) [inline townhall/myapi/dao.py:278]: I get the idea here with the dict, but this needs validation and/or structure such that someone doesnt try to pass in keys that dont make sense such as fields that dont exist. The use of dataclass to define this task_data param will be useful here. https://www.datacamp.com/tutorial/python-data-classes 

I see in types.py some have been added for the create operation, similar thing can be done here

File: townhall/myapi/models.py [hunk 4]
```diff
@@ -77,3 +77,22 @@ class Comment(models.Model):
...
+        ('in_progress', 'In Progress'),
+        ('completed', 'Completed'),
+    ]
+
+    name = models.CharField(max_length=255)
+    description = models.TextField()
+    deadline = models.DateTimeField()
+    status = models.CharField(max_length=20)
+    assigned_to = models.ForeignKey(Volunteer, on_delete=models.SET_NULL, null=True, related_name='assigned_tasks')
+    created_by = models.ForeignKey(Volunteer, on_delete=models.SET_NULL, null=True, related_name='created_tasks')
```
Comment (ansel-rangers11) [inline townhall/myapi/models.py:94]: fields look good

File: townhall/myapi/models.py [hunk 5]
```diff
@@ -77,3 +77,22 @@ class Comment(models.Model):
...
+    STATUS_CHOICES = [
+        ('open', 'Open'),
+        ('in_progress', 'In Progress'),
+        ('completed', 'Completed'),
+    ]
+
+    name = models.CharField(max_length=255)
+    description = models.TextField()
+    deadline = models.DateTimeField()
+    status = models.CharField(max_length=20)
```
Comment (ansel-rangers11) [inline townhall/myapi/models.py:92]: status can be defined using the choices as seen here https://docs.djangoproject.com/en/5.1/ref/models/fields/#enumeration-types

File: townhall/myapi/views.py [hunk 6]
```diff
@@ -223,4 +225,60 @@ def handle_organization_update(self, request):
...
+        task = TaskServices.get_task_by_id(pk)
+        if task:
+            return Response(TaskSerializer(task).data)
+        return Response({'error': 'Task not found'}, status=status.HTTP_404_NOT_FOUND)
+
+    @action(detail=False, methods=['post'])
+    def create_task(self, request, *args, **kwargs):
+      task_data = CreateTaskData(
+          name=request.data.get('name'),
+          description=request.data.get('description'),
```
Comment (ansel-rangers11) [inline townhall/myapi/views.py:251]: some of these may benefit from default fall back on get, so like `request.data.get('description', None)`

**Review by ansel-rangers11 — CHANGES_REQUESTED** [review]
Please take a look at feedback

File: townhall/myapi/views.py (hunk 6 above)
Comment (JordanBandur) [reply townhall/myapi/views.py]: Done

File: townhall/myapi/models.py (hunk 5 above)
Comment (JordanBandur) [reply townhall/myapi/models.py]: I made the modification

File: townhall/myapi/dao.py (hunk 3 above)
Comment (JordanBandur) [reply townhall/myapi/dao.py]: Made it more robust

File: townhall/myapi/dao.py (hunk 1 above)
Comment (JordanBandur) [reply townhall/myapi/dao.py]: Done

**Review by ansel-rangers11 — APPROVED** [review]

**Pull request merged**